opensearch: add selector to volumeClaimTemplates

Pale-whale opened the pull request:
Signed-off-by: Pierre Peloille <pierre.p@padoa-group.com>

### Description
This pr adds the ability to specify a selector in the `volumeClaimTemplate` for the opensearch chart.


### Issues Resolved
Fixes #239 

### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

prudhvigodithi commented on charts/opensearch/templates/statefulset.yaml:
```diff
@@ -43,6 +43,9 @@ spec:
     {{- end }}
     {{- end }}
+    {{- if .Values.persistence.selector }}
+    selector: "{{ .Values.persistence.selector }}"
```
Hey @Pale-whale this will actually cause the following error when adding `matchLabels:` for selector in values.yaml
` error validating data: ValidationError(StatefulSet.spec.volumeClaimTemplates[0].spec.selector)`

I would suggest to test by adding in statefulset as follows.
```
{{- if .Values.persistence.matchLabels }}
  selector:
    matchLabels:
{{ toYaml .Values.persistence.pvc.matchLabels | indent 12 }}
{{- end}}
```

prudhvigodithi replied:
@TheAlgo, @DandyDeveloper, @peterzhuamazon

TheAlgo replied:
Good point

peterzhuamazon commented:
Hi @Pale-whale would you address @prudhvigodithi comment?
Thanks.

Pale-whale commented:
Hello everyone, sorry I did not update this pr sooner.
I corrected the templating thanks to the suggestion of @prudhvigodithi and updated the Chart and the CHANGELOG to match version 1.10.
Let me know if I need to make any other changes, thanks :)

peterzhuamazon commented:
Thanks @Pale-whale 
@TheAlgo @prudhvigodithi please review again I will correct any version errors if any once you approve.
Thanks.

prudhvigodithi commented:
Hey @Pale-whale, just wanted to know if helm install of the OpenSearch chart worked for you with adding as ? 
 ```
 matchLabels:
     type: gp2
```
I see its breaking at 
`Error: INSTALLATION FAILED: YAML parse error on opensearch/templates/statefulset.yaml: error converting YAML to JSON: yaml: line 34: mapping values are not allowed in this context`

prudhvigodithi commented:
This error is actually misleading, more details I have added to the issue.
@Pale-whale

peterzhuamazon commented:
Hi @Pale-whale any updates based on @prudhvigodithi comments?
Thanks.

peterzhuamazon commented:
@Pale-whale you didnt sign DCO again I think.

prudhvigodithi commented:
This change will break the chart installation as mentioned in issue https://github.com/opensearch-project/helm-charts/issues/239 .

prudhvigodithi commented:
Hey @Pale-whale close this PR, please check the #239 for more details.
